### DISCUSSION TITLE
Why autocompletion for Avalonia doesn't work in VS Code on Linux

### DISCUSSION THREAD

**Discussion by al-zakaria** [AvaloniaUI/Avalonia#15820] [Q&A]
I  have downloaded extensions for c# and avalonia ui, but auto completion does not work.

while in .axaml.cs code everything works fine

I rebooted vs code and reinstalled but it doesn't help

**Comment by thevortexcloud** [comment]
I believe the VS code extension just adds the previewer and does not do much beyond that. For the record it does not work either in Visual Studio out of the box. Instead you need to use Resharper/Rider which do have proper Avalonia support. My advice would be to use Rider anyway on Linux if you can afford it, it's the only proper C# IDE available. You can also get it for free if you are a student.

The code behind files work because they are just standard C# files, which VS Code already handles.

**Reply by al-zakaria** [reply]
no, for example, in visual studio on my main computer, I wrote in Avalonia without ReSharper and everything worked fine and autocomplete worked fine.  and vs code on Windows everything also worked well and there was code completion with the usual extension, but nothing works on Linux.   and since at the moment I do not have access to my main computer, which works fine with all the heavy IDEs, I cannot use Rider.  and for the next 3 months I won’t be able to use my computer because I’m simply in another country. So I decided to use vs code since everything worked fine on my Windows computer
